tests/wallet/test_chialist.py

```diff
@@ -0,0 +1,75 @@
+from src.wallet.chialisp import (
+    sexp,
+    cons,
+    first,
+    rest,
+    nth,
+    args,
+    apply,
+    eval,
+    quote,
+    make_if,
+    make_list,
+    fail,
+    is_zero,
+)
+
+
+class TestChialisp:
+    def test_sexp(self):
+        assert sexp() == "()"
+        assert sexp(1) == "(1)"
+        assert sexp(1, 2) == "(1 2)"
+
+    def test_cons(self):
+        assert cons(1, 2) == "(c 1 2)"
+
+    def test_first(self):
+        assert first("(1)") == "(f (1))"
+
+    def test_rest(self):
+        assert rest("(1)") == "(r (1))"
+
+    def test_nth(self):
+        assert nth("val") == "val"
+        assert nth("val", 0) == "(f val)"
+        assert nth("val", 1) == "(f (r val))"
+        assert nth("val", 2) == "(f (r (r val)))"
+        assert nth("val", 2, 0) == "(f (f (r (r val))))"
+        assert nth("val", 2, 1) == "(f (r (f (r (r val)))))"
+        assert nth("val", 2, 2) == "(f (r (r (f (r (r val))))))"
+
```

Comment:
`nth` and `args` deserve some negative tests here.

```
def test_nth_negative_arguments(self):
    with pytest.raises(SomeError) as exception_info:
        nth("val", -1)
```

`nth("val", -1)`
`nth("val", 1,-1)`
`args(-1)`

Reply:
added